nix/tullia.nix

```diff
@@ -0,0 +1,123 @@
+self: system:
+
+let
+  inherit (self.inputs.nixpkgs) lib;
+  inherit (self.inputs.utils.lib) flattenTree;
+
+  ciInputName = "GitHub event";
+in rec {
+  tasks = let
+    common = {
+      config,
+      ...
+    }: {
+      preset = {
+        nix.enable = true;
+        github-ci = {
+          enable = config.actionRun.facts != {};
+          repo = "input-output-hk/cardano-node";
+          sha = config.preset.github-ci.lib.getRevision ciInputName null;
+          clone = false;
+        };
+      };
+    };
+
+    flakeUrl = {
+      config,
+      lib,
+      ...
+    }: lib.escapeShellArg (
+      if config.actionRun.facts != {}
+      then with config.preset.github-ci; "github:${repo}/${sha}"
+      else "."
```

Comment:
How come having a fact means you can refer to the flake with `.`?

Reply:
Tullia tasks can be run locally or on Cicero. When no facts are present we know that we are running locally and vice versa. When running locally, the current directory is already bind-mounted into the container, so we don't need to fetch the source  from GitHub.